modular_skyrat/modules/morgue_radio/morgue.dm

```diff
@@ -0,0 +1,36 @@
+/obj/structure/bodycontainer/morgue/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/morgue_radio)
+
+/obj/structure/bodycontainer
+	var/obj/item/radio/radio // The internal radio in the morgue trays.
+
+/datum/component/morgue_radio
+	var/obj/structure/bodycontainer/morgue/morgue // Typecasted reference to the current tray.
+
+
+/datum/component/morgue_radio/RegisterWithParent()
+	morgue = parent
+	morgue.radio = new /obj/item/radio(morgue) // Initialize the radio in the morgue tray
+	morgue.radio.set_listening(FALSE)
+	RegisterSignal(morgue, COMSIG_MORGUE_ALARM, .proc/morgue_revivable)
+
+/datum/component/morgue_radio/UnregisterFromParent()
+	QDEL_NULL(morgue.radio)
+	UnregisterSignal(morgue, COMSIG_MORGUE_ALARM)
+
+/datum/component/morgue_radio/proc/morgue_revivable(mob/living/cadaver)
```

Comment:
```suggestion
/datum/component/morgue_radio/proc/morgue_revivable(mob/living/cadaver)
	SIGNAL_HANDLER

```
It's a signal handler.

Reply:
signal handler sets should not sleep but yea its not 100 percent needed to work but ill do it anyways